source/apiVolontaria/volunteer/serializers.py

```diff
@@ -414,11 +415,39 @@ class ParticipationBasicSerializer(serializers.ModelSerializer):
 
     # Explicitly declare the BooleanField to make it "required"
     standby = serializers.BooleanField()
+    invit_icalandar = serializers.BooleanField(
+        default=False,
+        write_only=True,
+    )
     user = UserBasicSerializer(
         read_only=True,
         default=serializers.CurrentUserDefault(),
     )
 
+    def create(self, validated_data):
+        participation = models.Participation()
+        participation.standby = validated_data['standby']
+        participation.event = validated_data['event']
+        participation.user = validated_data['user']
```

Comment:
It's working but it could be great to use this kind of logic to be more generic and help in the future :

```python
    def create(self, validated_data):
        invitation_ics = validated_data.pop('invit_icalendar')
        participation = models.Participation.objects.create(**validated_data)
        # Add here your logic for ICS file
        return participation
```

Reply:
done

Reply:
The second line of my example `participation = models.Participation.objects.create(**validated_data)` can replace all your lines:
```python
        participation = models.Participation()
        participation.standby = validated_data['standby']
        participation.event = validated_data['event']
        participation.user = validated_data['user']

        ....

        participation.save()
```